modules/c-ares/1.19.1/patches/add_build_file.patch

```diff
@@ -0,0 +1,254 @@
+--- /dev/null
++++ BUILD.bazel
+@@ -0,0 +1,251 @@
++load("@bazel_skylib//rules:copy_file.bzl", "copy_file")
++
++config_setting(
++    name = "darwin",
++    values = {"cpu": "darwin"},
++)
++
++config_setting(
++    name = "darwin_x86_64",
++    values = {"cpu": "darwin_x86_64"},
++)
++
++config_setting(
++    name = "darwin_arm64",
++    values = {"cpu": "darwin_arm64"},
++)
++
++config_setting(
++    name = "darwin_arm64e",
++    values = {"cpu": "darwin_arm64e"},
++)
++
++config_setting(
++    name = "windows",
++    values = {"cpu": "x64_windows"},
++)
++
++# Android is not officially supported through C++.
++# This just helps with the build for now.
++config_setting(
++    name = "android",
++    values = {
++        "crosstool_top": "//external:android/crosstool",
++    },
++)
++
++# iOS is not officially supported through C++.
++# This just helps with the build for now.
++config_setting(
++    name = "ios_x86_64",
++    values = {"cpu": "ios_x86_64"},
++)
++
++config_setting(
++    name = "ios_armv7",
++    values = {"cpu": "ios_armv7"},
++)
++
++config_setting(
++    name = "ios_armv7s",
++    values = {"cpu": "ios_armv7s"},
++)
++
++config_setting(
++    name = "ios_arm64",
++    values = {"cpu": "ios_arm64"},
++)
++
++config_setting(
++    name = "ios_sim_arm64",
++    values = {"cpu": "ios_sim_arm64"},
++)
++
++# The following architectures are found in 
++# https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/rules/apple/ApplePlatform.java
++config_setting(
++    name = "tvos_x86_64",
++    values = {"cpu": "tvos_x86_64"},
++)
++
++config_setting(
++    name = "tvos_arm64",
++    values = {"cpu": "tvos_arm64"}
++)
++
++config_setting(
++    name = "watchos_i386",
++    values = {"cpu": "watchos_i386"},
++)
++
++config_setting(
++    name = "watchos_x86_64",
++    values = {"cpu": "watchos_x86_64"}
++)
++
++config_setting(
++    name = "watchos_armv7k",
++    values = {"cpu": "watchos_armv7k"},
++)
++
++config_setting(
++    name = "watchos_arm64_32",
++    values = {"cpu": "watchos_arm64_32"}
++)
++
++config_setting(
++    name = "openbsd",
++    values = {"cpu": "openbsd"},
++)
++
++copy_file(
++    name = "ares_build_h",
++    src = "@com_github_grpc_grpc//third_party/cares:ares_build.h",
++    out = "ares_build.h",
++)
++
++copy_file(
++    name = "ares_config_h",
++    src = select({
++        ":ios_x86_64": "@com_github_grpc_grpc//third_party/cares:config_darwin/ares_config.h",
```

Comment:
@veblush Do you know how those config files are generated? Ideally we should just generate them in this module instead of depending on grpc.

Reply:
Because this was set up many years ago, no one here is familiar with the details. I've done some digging and found a relevant GitHub PR (https://github.com/grpc/grpc/pull/7771).  It appears they used the `configure` command, which creates configuration files for the environment where it's executed.

I tried https://github.com/c-ares/c-ares/blob/main/ci/build.sh to see if I can do this on Cloudtop and it seems to work. Simply, I ran `autoreconf -fi` and `../configure` under a new build directory.

Reply:
OK, I think we can also just check in those config files as overlay files in BCR. This way, it doesn't have to depend on grpc repo or another external unhermetic tool. I also took the chance to simplify the BUILD file. PTAL.